utilities/transactions/lock/range/range_tree/lib/locktree/lock_request.cc

```diff
@@ -410,12 +410,13 @@ void lock_request::retry_all_lock_requests_info(lt_lock_request_info *info,
     // move on to the next lock request. otherwise
     // the request is gone from the list so we may
     // read the i'th entry for the next one.
-    r = request->retry(collector);
+    r = request->retry(&conflicts_collector);
     if (r != 0) {
       i++;
     }
   }
 
+  report_waits(&conflicts_collector, lock_wait_callback, callback_arg);
```

Comment:
Should we add a comment to `lock_wait_callback` to clarify that it will be called while holding a lock?

Reply:
I'll add a comment.